router/router.go

```diff
@@ -0,0 +1,208 @@
+package router
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pokt-foundation/transaction-db/types"
+	jsonresponse "github.com/pokt-foundation/utils-go/json-response"
+
+	"github.com/sirupsen/logrus"
+)
+
+type driver interface {
+	WriteError(ctx context.Context, poktErr types.Error) error
+	WriteSession(ctx context.Context, session types.PocketSession) error
+	WriteRegion(ctx context.Context, region types.PortalRegion) error
+	WriteRelay(ctx context.Context, relay types.Relay) error
+	ReadRelay(ctx context.Context, relayID int) (types.Relay, error)
+}
+
+type Router struct {
+	Router  *mux.Router
+	Driver  driver
+	APIKeys map[string]bool
+	log     *logrus.Logger
+}
+
+func (rt *Router) logError(err error) {
+	fields := logrus.Fields{
+		"err": err.Error(),
+	}
+
+	rt.log.WithFields(fields).Error(err)
+}
+
+// NewRouter returns router instance
+func NewRouter(driver driver, apiKeys map[string]bool, logger *logrus.Logger) (*Router, error) {
+	rt := &Router{
+		Driver:  driver,
+		Router:  mux.NewRouter(),
```

Comment:
With the mux project being no longer maintained and the repo archived, I think is best to use another routing library, chi is a good lightweight alternative